pkg/types/request_test.go

```diff
@@ -200,6 +201,21 @@ func TestRequestStringRepresentations(t *testing.T) {
 		_, err := ParseProviderStrings("http://127.0.0.1:5000/nope")
 		require.ErrorContains(t, err, "paths not supported")
 	})
+
+	t.Run("fixed peer, protocol included", func(t *testing.T) {
+		pps, err := ParseProviderStrings("/ip4/127.0.0.1/tcp/5000/p2p/12D3KooWBSTEYMLSu5FnQjshEVah9LFGEZoQt26eacCEVYfedWA4+bitswap")
```

Comment:
some multiaddr aficionado is going to find this one day and get all twitchy

Reply:
some day, but not this day :P